app/models/operatingsystem.rb

```diff
@@ -255,8 +255,18 @@ def boot_filename(host = nil)
     return default_boot_filename if host.nil? || host.pxe_loader.nil?
     return host.foreman_url('iPXE') if host.pxe_loader == 'iPXE Embedded'
     architecture = host.arch.nil? ? '' : host.arch.bootfilename_efi
-    boot_uri = URI.parse((host.subnet&.httpboot?) ? host.subnet.httpboot.url : Setting[:unattended_url])
-    self.class.all_loaders_map(architecture, "#{boot_uri.host}:#{boot_uri.port}")[host.pxe_loader]
+    if host.subnet&.httpboot?
+      if host.pxe_loader =~ /UEFI HTTPS/
+        port = host.subnet.httpboot.setting(:HTTPBoot, 'https_port') || 8443
```

Comment:
Since it differs for Katello I'm wondering if we should try to detect this based on the `host.subnet.httpboot.url`. If its scheme is https then we can use that port as a default. On the other hand, that's rather complex and I'd also be fine with hard requiring a proxy that exposes this setting which would mean a hard exception + a note in the upgrade notes. Slightly leaning to that and avoid the guessing.

Reply:
What makes you believe that user uses HTTPS for smart proxy API communication? For example in my development setup I use HTTP so I can track the communication using wireshark.

Reply:
I'd rather hard error if we can't come up with a sane default than try to guess. As an admin that's easier to debug. So essentially `host.subnet.httpboot.setting(:HTTPBoot, 'https_port') || raise`

Reply:
Ok I can do that.